### PR TITLE
Feature/tf df serving

### PR DESCRIPTION
**Pull request by picousse**
Basic tip on the TF-DF TF Serving docker container

### CODE REVIEW

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Serving TF-DF models using TF Serving
+
+*This is a small excerpt from the  https://blog.ml6.eu/serving-decision-forests-with-tensorflow-b447ea4fc81c blogpost*
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:3]:
```suggestion
*This is a small excerpt from our [blogpost on decision forests](https://blog.ml6.eu/serving-decision-forests-with-tensorflow-b447ea4fc81c)*
```

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+TensorFlow and TensorFlow Extended are a broad ecosystem of software packages, allowing to train, serve,... ML models.
+
+The addition of decision forests to TensorFlow (blogpost)[https://www.tensorflow.org/decision_forests], increases the applicability of TensorFlow on many more use cases.
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:7]:
```suggestion
The [addition of decision forests to TensorFlow](https://www.tensorflow.org/decision_forests) increases the applicability of TensorFlow on many more use cases.
```

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Training is only one part of ML models: robust inference calculations are as important in a data pipeline.
+This is where TensorFlow Serving comes in. TensorFlow Serving basically creates an API around a model. On top of that it allows to split code and model, enabling separate versioning.
+
+Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to except these new models.
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:12]:
```suggestion
Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to serve these new models.
```

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+As we found TF-DF models to be very interesting, we, at ML6, decided to build a custom docker image to serve these models.
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:14]:
```suggestion
We decided to build a custom `tf-serving` docker image to serve these models.
```

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+The docker image can be found at (https://hub.docker.com/r/ml6team/tf-serving-tfdf)[https://hub.docker.com/r/ml6team/tf-serving-tfdf].
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:16]:
```suggestion
The docker image can be found on [DockerHub](https://hub.docker.com/r/ml6team/tf-serving-tfdf).
```

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+To use the docker image for a penguin model, the following code can be used:
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:18]: Maybe refer to the colab that was used to train this penguin model?

**Review by julienschuermans — COMMENTED** [review]
Nice work, I've fixed some hyperlinks and added a simple comment.
Can you also add a line to the `quick_tips/structured_data/README.md` with a reference to this new `2021_07_02_TFDF_serving` directory?

Sorry for the late response!

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md (hunk 6 above)
Comment (picousse) [reply structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:19]: has been added

**Comment by picousse** [comment]
@julienschuermans I think I applied all your changes, added the colab link and added the link.
Ready to be merged if you ask me.

Do you want to check before merging?

File: structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+Training is only one part of ML models: robust inference calculations are as important in a data pipeline.
+This is where TensorFlow Serving comes in. TensorFlow Serving basically creates an API around a model. On top of that it allows to split code and model, enabling separate versioning.
+
+Unfortunately, when TensorFlow Decision Forests (TF-DF) was released, TensorFlow Serving was not ready to serve these new models.
+
+We decided to build a custom `tf-serving` docker image to serve these models.
+
+The docker image can be found on [DockerHub](https://hub.docker.com/r/ml6team/tf-serving-tfdf).
+The model used in the following example was trained using the [TF-DF pinguins colab](https://colab.research.google.com/github/tensorflow/decision-forests/blob/main/documentation/tutorials/beginner_colab.ipynb).
```
Comment (julienschuermans) [inline structured_data/2021_07_02_TFDF_serving/TF-DF-serving.md:17]:
```suggestion
The model used in the following example was trained using the [TF-DF penguins colab](https://colab.research.google.com/github/tensorflow/decision-forests/blob/main/documentation/tutorials/beginner_colab.ipynb).
```

File: structured_data/README.md [hunk 8]
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
 This document contains an overview of the currently available labeling tools for time series. What tools are the ones to be used, which ones you probably better avoid.
 - [_Tabular Data: Tensorflow Data Validation_ (30/06/2021)](2021_06_30_tensorflow_data_validation):
 This notebook showcases the functionality of TensorFlow Data Validation (TFDV), a tool developed by tensorflow
-to track anomalous changes in your data.
\ No newline at end of file
+to track anomalous changes in your data.
+- [_TF Decision Forest: Serving TF-DF models with TF Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
```
Comment (julienschuermans) [inline structured_data/README.md:16]:
```suggestion
- [_Tabular Data: Serving decision forests with Tensorflow Serving_ (02/07/2021)](2021_07_02_TFDF_serving):
```

File: structured_data/README.md [hunk 9]
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
...
+  This Readme documents the use of the custom build TF Serving docker build to serve TF-Decision Forest
```
Comment (julienschuermans) [inline structured_data/README.md:17]:
```suggestion
  This Readme documents the use of our custom TF Serving docker image built to serve Tensorflow Decision Forests.
```

File: structured_data/README.md [hunk 10]
```diff
@@ -12,4 +12,7 @@ This notebook provides looks into how MINIROCKET — a novel feature extraction
...
+  models.
```
Comment (julienschuermans) [inline structured_data/README.md:18]:
```suggestion
```

**Comment by julienschuermans** [comment]
LGTM, let's merge it!

**Pull request merged**